sd_mecha/merge_methods.py

```diff
@@ -133,15 +141,26 @@ def add_cosine_generic(a: Tensor, b: Tensor, alpha: float, similarity: Tensor) -
 # latex notes in reference to original implementation: https://arxiv.org/abs/2306.01708
 # - `delta`: $$ \hat{\tau}_t $$
 # - `signs`: $$ \gamma_t $$
-# - `final_sign`: $$ \gamma_m^p = sgn(\sum_{t=1}^n \hat{\tau}_t^p) $$
+# - `final_sign`: $$ \gamma_m^p = sgn(\Sigma_{t=1}^n \hat{\tau}_t^p) $$
 # - `delta_filters`: $$ \{ \gamma_t^p = \gamma_m^p \} $$
 # - `param_counts`: $$ |A^p| $$
-# - `filtered_delta`: $$ \sum_{t\in{A^p}} \hat{\tau}_t^p $$
+# - `filtered_delta`: $$ \Sigma_{t\in{A^p}} \hat{\tau}_t^p $$
 # - `return`: $$ \lambda * \tau_m $$
+# Special mode "TIES-SOUP" has been implemented by setting `vote_sgn` > 0.0
+# - `final_sign`: $$ \gamma_m^p = sgn(\Sigma_{t=1}^n \gamma_t^p) $$
+# Special mode "TIES-STOCK" has been implemented by setting `apply_stock` > 0.0
+# Special mode "TIES-GMEDIAN" has been implemented by setting `apply_median` > 0.0
 @convert_to_recipe
 def ties_sum(  # aka add_difference_ties
     *models: Tensor | LiftFlag[MergeSpace.DELTA],
     k: Hyper = 0.2,
+    vote_sgn: Hyper = 0.0,
+    apply_stock: Hyper = 0.0,
+    cos_eps: Hyper = 1e-6,
+    apply_median: Hyper = 0.0,
+    eps: Hyper = 1e-6,    
+    maxiter: Hyper = 100, 
+    ftol: Hyper =1e-20,
```

Comment:
One thing to consider here is recipes that are already serialized and existing user code. If there are any comfyui workflows that use `ties_sum` from comfy-mecha for example, then the existing nodes will not see these new inputs added to the node. This is not too much a problem as long as the default values would preserve the original behavior without the parameters. Can you confirm this is the case?

As a side note, I am generally in favor of writing new merge methods instead of adding parameters to existing ones. Do you think it is possible to put the new code in a separate fundamental merge method? I wonder how practical that would be. All of the new code, except for the `vote_sgn` parameter, seems to be located at the end of the method, so maybe it is? Let me know if you think it is possible.

Reply:
> In favor of writing new merge methods

I have made a [research log](https://github.com/6DammK9/nai-anime-pure-negative-prompt/blob/main/ch05/README_XL.MD#findings-on-sdxl-ties) for such SDXL merge. The following suggestion (split to 3 algos with no stacking) is up to you:
- First, make a set of **original** TIES merge, then "TIES w/ DARE" as DARE, and try to figure out how to implement DARE on top of a known receipt. It may work for some (or most?) cases as [implemented in other repos](https://github.com/54rt1n/ComfyUI-DareMerge/blob/master/components/dare.py#L15), but it is not enough for me for my [Pony x SDXL merge](https://civitai.com/models/309514?modelVersionId=484834). It can be just a monolithic method with union of parameters.
- Then, obviously Geometric merge can be implemented alone, and I expect it behaves like `weighted_sum` but closer to base model.
- Finally implement this TGMD merge **exclusively**. Since I have already [overrided more then half of the original algorithm](https://github.com/6DammK9/nai-anime-pure-negative-prompt/blob/main/ch01/modelstock.md#spinoff-model-stonk-by-calculate-geometric-median-with-gradient-descent), I think it is suitable to trim all the parameters out, leaving only `p=0.1` for the dropout part. Even `maxiter` / `eps` / `ftol` are _safe to hide from user's sight._ Even `k=1.0` / `alpha=1.0` are in the same case: They _defines_ TGMD merge as an algorithm.

Since I'm the minority who [use dedicated python notebook](https://github.com/6DammK9/nai-anime-pure-negative-prompt/blob/main/ch05/astolfo_mix_e2e.ipynb) and a [ridiculous workstation build](https://github.com/6DammK9/nai-anime-pure-negative-prompt/blob/main/ch04/ice_lake_ws.md), I always able to find a way to keep my TGMD merge after breaking changes accoring to ComfyUI adaption, so feel free to make your decision. To get users well informed, we can specify that such memory intensive (O(N) instead of O(1), which is still harsh to me) algoithm will be forced to use `device='cpu'` to prevent OOM.

Reply:
> Since I'm the minority who [use dedicated python notebook](https://github.com/6DammK9/nai-anime-pure-negative-prompt/blob/main/ch05/astolfo_mix_e2e.ipynb) and a [ridiculous workstation build](https://github.com/6DammK9/nai-anime-pure-negative-prompt/blob/main/ch04/ice_lake_ws.md), I always able to find a way to keep my TGMD merge after breaking changes accoring to ComfyUI adaption, so feel free to make your decision.

Got it. I appreciate the cordial and constructive approach to the conversation. I'll try to break the code into smaller recipes and then we can merge.